llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2756,6 +2756,96 @@ LogicalResult WinogradOutputTransformOp::reifyResultShapes(
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+//===----------------------------------------------------------------------===//
+// SoftmaxOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult SoftmaxOp::verify() {
+  Operation *op = getOperation();
+  auto inputType = input().getType().cast<ShapedType>();
+  auto outputType = output().getType().cast<ShapedType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+  ArrayRef<int64_t> outputShape = outputType.getShape();
+  if (!areShapesCompatible(inputShape, outputShape)) {
+    return op->emitOpError("incompatible output shape");
+  }
+  int64_t inputRank = getInputOperandRank();
+  int64_t dimension = getDimension();
+  if ((dimension < 0) || (dimension >= inputRank)) {
+    return op->emitOpError("incorrect dimension specified");
+  }
+  return success();
+}
+
+SmallVector<Range> SoftmaxOp::getIterationDomain(OpBuilder &builder) {
+  int64_t operandRank = getInputOperandRank();
```

Comment:
Something interesting here (not necessarily for this PR), this seems to represent the iteration space of the entire operation, which is a bit hard to reason about in general. Seems like its basically shape of the output which in this case is same as the shape of the input... Something to keep track off as we use this path more.